website/src/components/visualisation/Resources/LineGraph.tsx

```diff
@@ -0,0 +1,145 @@
+            datapaths: {
+                team1: "ClientInfos.Team1.Resources", 
+                team2: "ClientInfos.Team2.Resources", 
+                team3: "ClientInfos.Team3.Resources", 
+                team4: "ClientInfos.Team4.Resources", 
+                team5: "ClientInfos.Team5.Resources", 
+                team6: "ClientInfos.Team6.Resources", 
+            },
+            chartData: outputJSON
+        };
+    }
+
+    handleClick = dataKey => {
+        if(_.includes(this.state.disabled, dataKey)) {
+            this.setState({
+                disabled: this.state.disabled.filter(obj => obj !== dataKey)
+            });
+        }
+        else {
+            this.setState({
+                disabled:this.state.disabled.concat(dataKey)
+            });
+        }
+    };
+
+    renderCustomizedLegend = ({ payload }) => {
+        return(
+            <div className="customized-legend">
+                {payload.map(entry => {
+                    const { dataKey, colour } = entry;
+                    const active = _.includes(this.state.disabled, dataKey);
+                    const style = {
+                        marginRight: 10, 
+                        colour: active ? "#AAA" : "#000"
+                    };
+
+                    return(
+                        <span
+                            className="legend-item" 
+                            onClick={() => this.handleClick(dataKey)}
+                            style={style}
+                        >
+                            <Surface width={10} height={10} viewBox="0 0 10 10" >
+                                <Symbols cx={5} cy={5} type="circle" size={50} fill={colour} />
+                                {active && (
+                                    <Symbols 
+                                        cx={5}
+                                        cy={5}
+                                        type="circle"
+                                        size={25}
+                                        fill={"#FFF"}
+                                    />
+                                )}
+                            </Surface>
+                            <span>{dataKey}</span>
+                        </span>
+                    )
+                })}
+            </div>
+        );
+    };
+
+    getSeasonEnds () {
+        var seasonEnds: any[] = [];
+        let i;
+        for(i in this.state.chartData.GameStates) {
+            if(this.state.chartData.GameStates[i].Environment.LastDisasterReport.Magnitude !== 0){
+                seasonEnds.push(this.state.chartData.GameStates[i].Turn -1)
+            }
+        }
+        return seasonEnds;
+    };
+
+    render() {
+        return (
+            <ResponsiveContainer height={330} width = "95%">
+                <LineChart data={this.state.chartData.GameStates}
+                    margin={{ top: 5, right: 30, left: 20, bottom: 5 }}>
+                    {_.toPairs(this.state.lineColours)
```

Comment:
hmm -- is this indented ok to you? also it looks to have a lot of logic, so we could make this a class fn

Reply:
@olly-larkin i think your diff can benefit from` _.toPairs`, seems neat

Reply:
Do you mean a class function for LineChart? Or a class function the Line lines in the chart?

Reply:
And the tabbing in those four line 98-101 are right but the tabbing below was wrong. I did the autoformat  thing so should all be tabbed properly now

Reply:
> hmm -- does this look indented ok to you? also it looks to have a lot of logic, so we could make this a class fn

The indenting is right for 98-101 but was wrong below that - I did the autoformat thing so should all be fine now. Do you mean a class fn for LineChart or just the Line lines?

Reply:
the lines from 101 onwards